components/eamxx/src/diagnostics/zonal_avg.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef EAMXX_ZONAL_AVERAGE_HPP
+#define EAMXX_ZONAL_AVERAGE_HPP
+
+#include "share/atm_process/atmosphere_diagnostic.hpp"
+
+namespace scream {
+// TODO: Update this comment
+/*
+ * This diagnostic will calculate the area-weighted average of a field
+ * across the COL tag dimension, producing an N-1 dimensional field
+ * that is area-weighted average of the input field.
+ */
+
+class ZonalAvgDiag : public AtmosphereDiagnostic {
+
+  // TODO: comment this, noting it's a utility function that could exist elsewhere
```

Comment:
You may consider moving this fcn into `share/field/field_utils.hpp`. That's where we also have `horiz_contraction`, and other utilities. Not sure if we will ever use it outside of this diag class, but since it has no dependency on the class internals (it's a static method), we may as well put it out there...